### PR TITLE
Issues/#243 Add blocktimes.py to script library

### PR DESCRIPTION
**Pull request by macterra**
Generates a csv file like:

```
    1, 2018-12-08 00:41:44, 1064325,     0.00
    2, 2018-12-08 00:50:00,   496,     0.00
    3, 2018-12-08 00:50:06,     6,     0.00
    4, 2018-12-08 00:50:08,     2,     0.00
    5, 2018-12-08 00:50:20,    12,     0.00
    6, 2018-12-08 00:50:44,    24,     0.00
    7, 2018-12-08 00:51:07,    23,    80.43
    8, 2018-12-08 00:51:31,    24,    13.00
    9, 2018-12-08 00:53:08,    97,    26.00
   10, 2018-12-08 00:53:22,    14,    27.71

...

  297, 2018-12-10 11:16:24,  1624,   403.14
  298, 2018-12-10 11:30:04,   820,   497.14
  299, 2018-12-10 11:38:11,   487,   527.71
  300, 2018-12-10 11:48:09,   598,   536.71
  301, 2018-12-10 11:51:14,   185,   535.00
  302, 2018-12-10 11:58:22,   428,   591.71
  303, 2018-12-10 12:05:00,   398,   416.57
  304, 2018-12-10 12:20:50,   950,   435.14
  305, 2018-12-10 12:29:34,   524,   440.43
```

### CODE REVIEW

**Comment by macterra** [comment]
Suitable for charting in Excel or Google Sheets...

image

File: eqb/scripts/blocktimes.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Generates a csv of blocktimes suitable for importing into Excel or Google Docs
+#
+# Depends on https://github.com/jgarzik/python-bitcoinrpc
+# pip install python-bitcoinrpc
+
+from bitcoinrpc.authproxy import AuthServiceProxy, JSONRPCException
+from datetime import datetime
+
```
Comment (ealefad) [inline eqb/scripts/blocktimes.py:8]: I have a suggestion to move node address/port and RPC credentials to the command line, we can add this line here:
`from sys import argv, exit`

File: eqb/scripts/blocktimes.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Generates a csv of blocktimes suitable for importing into Excel or Google Docs
+#
+# Depends on https://github.com/jgarzik/python-bitcoinrpc
+# pip install python-bitcoinrpc
+
+from bitcoinrpc.authproxy import AuthServiceProxy, JSONRPCException
+from datetime import datetime
+    
+# rpc_user and rpc_password are set in the bitcoin.conf file
+rpc_connection = AuthServiceProxy("http://equibit:equibit@127.0.0.1:18333")
```
Comment (ealefad) [inline eqb/scripts/blocktimes.py:10]: and replace line 10 with:
```
if len(argv) != 4:
    print("Usage: python blocktimes.py nodeAddress:rpcPort rpcUser rpcPassword")
    exit(1)

rpcAddr = argv[1]
rpcUser = argv[2]
rpcPass = argv[3]
rpc_connection = AuthServiceProxy("http://{}:{}@{}".format(rpcUser, rpcPass, rpcAddr))
```

**Review by ealefad — APPROVED** [review]

File: eqb/scripts/blocktimes.py (hunk 2 above)
Comment (r-soltani) [reply eqb/scripts/blocktimes.py:10]: I agree. This would be very helpful.
Already added to my update

File: eqb/scripts/blocktimes.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    block_time = block_times[i]
+    hash_time = block_time - block_times[i-1]
+    timestamp = datetime.utcfromtimestamp(block_time).strftime('%Y-%m-%d %H:%M:%S')
+    if i > 6:
+        window = block_times[i-6:i+1]
+        time_avg = average_diff(window)
+    else:
+        time_avg = 0
+        
+    print('{:>5}, {}, {:>5}, {:8.2f}'.format(i, timestamp, hash_time, time_avg))
```
Comment (r-soltani) [inline eqb/scripts/blocktimes.py:40]: We can perform all the visualization using Python.
I will add the update I made to this PR, performing  regression and visualization on the data.

**Review by r-soltani — APPROVED** [review]

File: eqb/scripts/blocktimes.py (hunk 2 above)
Comment (macterra) [reply eqb/scripts/blocktimes.py:10]: Excellent idea, I took the liberty of opening #247 to track it.

File: eqb/scripts/blocktimes.py (hunk 3 above)
Comment (macterra) [reply eqb/scripts/blocktimes.py:40]: Please open another PR for that.

**Pull request merged**